contracts/HeapOrdering.sol

```diff
@@ -237,8 +240,10 @@ library HeapOrdering {
         uint256 accountsLength = _heap.accounts.length;
 
         // Swap the last account and the account to remove, then pop it.
-        swap(_heap, index, accountsLength - 1);
-        if (_heap.size == accountsLength) _heap.size--;
+        unchecked {
+            swap(_heap, index, accountsLength - 1);
+            if (_heap.size == accountsLength) _heap.size--;
+        }
```

Comment:
> Looks nice ! Would love to see the gas comparison. For that it is problematic that there are 2 kinds of optimizations:
> 
> * the uncheckeds
> * the removal of the `rightChildIndex` variable
> 
> And because of this we can't be sure what would be causing the gas consumption decrease

I updated the PR with gas report.


For the 2 optimizations, I roll back my change with rightChildIndex and add the "global" unchecked. This gives me that (left is the current PR, right is `rightChildIndex` + global unchecked):

<img width="1761" alt="Screenshot 2022-09-20 at 13 47 17" src="https://user-images.githubusercontent.com/44097430/191249770-147e4df7-08fc-456a-866c-263715d5dbcc.png">

Reply:
Current PR removes the `rightChildIndex`, so the changes on the right seem better to me: it's more efficient and a bit easier to read IMO with `rightChildIndex`

Reply:
I also think this particular "unchecked" should be removed (only for remove)